Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h

```diff
@@ -531,6 +557,7 @@ typedef NS_ENUM(NSInteger, PKPaymentButtonType) {
 
 NS_ASSUME_NONNULL_BEGIN
 
+#if !HAVE(PASSKIT_PAYMENT_ORDER_DETAILS)
```

Comment:
i dont think this is the right change, as `PKPaymentAuthorizationResult` is not dependent on `HAVE(PASSKIT_PAYMENT_ORDER_DETAILS)`.  it predates it

what exactly is the error that is necessitating this change?

Reply:
`#import <PassKit/PKPaymentRequestStatus.h>` on L78 is gated by `HAVE(PASSKIT_PAYMENT_ORDER_DETAILS)`, and it pulls PassKit's declaration of these classes. So I wanted to continue to declare our SPI when PASSKIT_PAYMENT_ORDER_DETAILS is off (effectively, in Monterey / iOS 15 and below).

Looking back at this, though, it seems like these types may have been public API in previous releases, and we can clean them up further.